packages/upload-client/src/index.js

```diff
@@ -129,7 +130,27 @@ async function uploadBlockStream(conf, blocks, options = {}) {
           const multihashDigest = await hasher.digest(bytes)
           /** @type {import('@web3-storage/capabilities/types').PieceLink} */
           const piece = Link.create(raw.code, multihashDigest)
+
+          // Invoke store/add and write bytes to write target
           const cid = await Store.add(conf, bytes, options)
+          // Invoke filecoin/offer for data
+          const result = await Storefront.filecoinOffer(
+            {
+              issuer: conf.issuer,
+              audience: conf.audience,
+              // Resource of storefront is the issuer did
+              with: conf.issuer.did(),
```

Comment:
Not sure I understand why this is the issuer?

https://github.com/web3-storage/w3up/blob/f5bac9d27610d976d1144ea12a5c6f9448e7974a/packages/capabilities/src/filecoin/storefront.js#L22-L25

lolz looks like I wrote that comment...still why?

Reply:
my comment is actually also wrong, the did must be the same as the issuer. Otherwise you would need to either have been delegated such capability or inline proofs given resource is not did of issuer.